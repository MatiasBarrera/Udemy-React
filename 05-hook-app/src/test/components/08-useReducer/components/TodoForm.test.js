import { shallow } from 'enzyme'
import React from 'react'
import TodoForm from '../../../../components/08-useReducer/components/TodoForm'

describe('Pruebas en TodoForm', () => {
  const handleAddTodo = jest.fn()
  const wrapper = shallow(<TodoForm handleAddTodo={handleAddTodo} />)

  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('No debe de llamar handleAddTodo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit')
    formSubmit({ preventDefault() {} })

    expect(handleAddTodo).toHaveBeenCalledTimes(0)
  })

  test('Debe de llamar la funcion handleAddTodo', () => {
    const value = 'Aprender react'
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description'
      }
    })
    const formSubmit = wrapper.find('form').prop('onSubmit')
    formSubmit({ preventDefault() {} })

    expect(handleAddTodo).toHaveBeenCalledTimes(1)
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object))
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false
    })
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
