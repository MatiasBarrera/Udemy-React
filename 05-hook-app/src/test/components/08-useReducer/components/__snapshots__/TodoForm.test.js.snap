// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoForm Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 w-100"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
